Add TravisCI Integration

ayaleloehr opened the pull request.

ayaleloehr commented:
After about 12 commits trying to get this darn thing to work on TravisCI (issues with installing QT dependencies for capybara-webkit mostly), this is ready for review. @plusjeff can you do the honors?

plusjeff commented on .travis.yml:
```diff
@@ -0,0 +1,26 @@
+
+language: ruby
+rvm:
+  - ruby-2.2.3
```
Instead of adding this, let's add a `.ruby-version` file. Travis will look at that, but it'll also make it easier to be in the correct Ruby version on the development side.

plusjeff commented on .travis.yml:
```diff
@@ -0,0 +1,26 @@
+
+install:
+  - bundle install
+  - export RAILS_ENV=test
```
This line can get removed:
https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables

plusjeff commented on .travis.yml:
```diff
@@ -0,0 +1,26 @@
+  - export RAILS_ENV=test
+  - bundle exec rake db:create
+  - bundle exec rake db:schema:load
+  - bundle exec rake db:migrate RAILS_ENV=test
```
Can remove the RAILS_ENV:
https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables

plusjeff commented on .travis.yml:
```diff
@@ -0,0 +1,26 @@
+  - bundle exec rake db:migrate RAILS_ENV=test
+  
+script:
+  - xvfb-run -a bundle exec rake
```
What does the `xvfb-run -a` do?

ayaleloehr commented:
@plusjeff Addressed your comments with commits. Only one not addressed via commit was your question on `xvfb-run -a`.  I used this is because VEC uses the capybara-webkit gem, which requires an X server to run, although it doesn't create any visible windows.  The [capybara-webkit documentation](https://github.com/thoughtbot/capybara-webkit#ci) says this is the correct way to make this work. That make sense?

plusjeff commented:
Cool - totally makes sense!
